cmake/Options/TileDBToolchain.cmake

```diff
@@ -33,11 +33,20 @@ if (NOT DEFINED CMAKE_TOOLCHAIN_FILE)
     endif()
 endif()
 
-if(APPLE AND NOT DEFINED VCPKG_TARGET_TRIPLET)
-    if (CMAKE_OSX_ARCHITECTURES STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR MATCHES "(x86_64)|(AMD64|amd64)|(^i.86$)")
-        set(VCPKG_TARGET_TRIPLET "x64-macos")
-    elseif (CMAKE_OSX_ARCHITECTURES STREQUAL arm64 OR CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "^arm")
-        set(VCPKG_TARGET_TRIPLET "arm64-macos")
+if(NOT DEFINED VCPKG_TARGET_TRIPLET AND SANITIZER STREQUAL "address")
```

Comment:
This seems like an odd way of getting these triplets defined. Is there some way of using the triplet detection that's in vcpkg already? If we could call that, get a triplet then all we would need is to append `"-asan"` to it.

Reply:
> Is there some way of using the triplet detection that's in vcpkg already?

Unfortunately not. [The logic is laid out](https://github.com/microsoft/vcpkg/blob/c75ff8cf3b784f67aa614552f32a6b2c7b9d8efc/scripts/buildsystems/vcpkg.cmake#L233-L369) in `vcpkg.cmake` without an enclosing function.